src/xtgeo/grid3d/grid_properties.py

```diff
@@ -1,24 +1,106 @@
 logger = xtg.functionlogger(__name__)
 
 
+def gridproperties_from_file(
+    pfile,
+    grid,
+    fformat=None,
+    names=None,
+    dates=None,
+    namestyle=0,
+    strict=(True, False),
+):
+    """Import grid properties from file.
+
+    In case of names='all' then all vectors which have a valid length
+    (number of total or active cells in the grid) will be read
+
+    Args:
+        pfile (str or Path): Name of file with properties
+        fformat (str): roff/init/unrst
+        names: list of property names, e.g. ['PORO', 'PERMX'] or 'all'
+        dates: list of dates on YYYYMMDD format, for restart files, or 'all'
+        grid (obj): The grid geometry object (optional if ROFF)
+        namestyle (int): 0 (default) for style SWAT_20110223,
+            1 for SWAT--2011_02_23 (applies to restart only)
+        strict (tuple of (bool, bool)): If (True, False) (default) then an
+            Error is raised if keyword name is not found, or a key-date combination
+            is not found. However, the dates will processed so that non-valid dates
+            are skipped (still, invalid key-date combinations may occur!).
+            If (True, True) all keywords and dates are tried, while (False, False)
+            means that that only valid entries are imported, more or less silently.
+            Saturations keywords SWAT/SOIL/SGAS are not evaluated as they may be
+            derived.
+    Example::
+        >>> grd = xtgeo.grid_from_file(reek_dir + "/REEK.EGRID", fformat='egrid')
+        >>> gps = xtgeo.gridproperties_from_file(
+        ...     reek_dir + "/REEK.INIT",
+        ...     fformat='init',
+        ...     names=["PORO", "PERMX"],
+        ...     grid=grd,
+        ... )
+    """
+    pfile = xtgeo._XTGeoFile(pfile, mode="rb")
+
+    pfile.check_file(raiseerror=ValueError)
+
+    if fformat is None or fformat == "guess":
+        fformat = pfile.detect_fformat()
+    else:
+        fformat = pfile.generic_format_by_proposal(fformat)  # default
+
+    if fformat.lower() in ["roff_ascii", "roff_binary"]:
+        return GridProperties(
+            props=[
+                GridProperty(pfile, fformat="roff", name=name, grid=grid)
+                for name in names
+            ]
+        )
+
+    elif fformat.lower() == "init":
+        return GridProperties(
+            props=_gridprops_import_eclrun.import_ecl_init_gridproperties(
+                pfile,
+                grid=grid,
+                names=names,
+                strict=strict[0],
+            )
+        )
+    elif fformat.lower() == "unrst":
+        return GridProperties(
+            props=_gridprops_import_eclrun.import_ecl_restart_gridproperties(
+                pfile,
+                dates=dates,
+                grid=grid,
+                names=names,
+                namestyle=namestyle,
+                strict=strict,
+            )
+        )
+    else:
+        raise ValueError("Invalid file format {fformat}")
+
+
```

Comment:
Cannot see "asmasked" in param-list

Reply:
Its in the list of parameters to `get_actnum` and `get_ijk`. Perhaps move the comment even further down?

Reply:
Yeah...  'activeonly' is in paramlist, so I got a little confused. Comment should avoid such possible confusions

Reply:
True, I think this is a separate issue as this notice occurrs more than one place so outside the scope of the PR. I simply moved it to before the first occurrance of something relevant in the file.

Reply:
Perhaps add something along the lines of

`Note that 'asmasked' and 'activeonly' are just different names for the same flag when passed
to different methods.
`

to the comment?

Reply:
I think I would have to look at that separately from this PR to get a good solution to those comments.

Reply:
ok - create issue for that?

Reply:
https://github.com/equinor/xtgeo/issues/729